test/celeritas/optical/Rayleigh.test.cc

```diff
@@ -0,0 +1,123 @@
+{
+using namespace ::celeritas::test;
+
+//---------------------------------------------------------------------------//
+
+class RayleighInteractorTest : public InteractorHostTestBase
+{
+  protected:
+    void SetUp() override
+    {
+        // Check incident quantities are valid
+        this->check_direction_polarization(
+            this->direction(), this->photon_track().polarization());
+    }
+
+    void sanity_check(Interaction const& interaction) const
+    {
+        // Interactions should always be scattering
+        EXPECT_EQ(Interaction::Action::scattered, interaction.action);
+        this->check_direction_polarization(interaction);
+    }
+};
+
+TEST_F(RayleighInteractorTest, basic)
+{
+    int const num_samples = 4;
+
+    RayleighInteractor interact{this->photon_track(), this->direction()};
+
+    auto& rng_engine = this->rng();
+
+    std::vector<real_type> dir_angle;
+    std::vector<real_type> pol_angle;
+
+    for ([[maybe_unused]] int i : range(num_samples))
+    {
+        Interaction result = interact(rng_engine);
+        this->sanity_check(result);
+
+        dir_angle.push_back(dot_product(result.direction, this->direction()));
+        pol_angle.push_back(dot_product(result.polarization,
+                                        this->photon_track().polarization()));
+    }
+
+    static real_type const expected_dir_angle[] = {-0.38366589898599,
+                                                   -0.18253767807648,
+                                                   0.42140775018143,
+                                                   -0.15366976713254};
+    static real_type const expected_pol_angle[] = {
+        0.46914077892327, 0.31484033374691, 0.82815845476385, 0.50516788333603};
+
+    EXPECT_VEC_SOFT_EQ(expected_dir_angle, dir_angle);
+    EXPECT_VEC_SOFT_EQ(expected_pol_angle, pol_angle);
+}
+
+TEST_F(RayleighInteractorTest, stress_test)
+{
+    int const num_samples = 1'000;
+
+    RayleighInteractor interact{this->photon_track(), this->direction()};
+
+    auto& rng_engine = this->rng();
+
+    real_type dir_angle_1st_moment = 0;
+    real_type dir_angle_2nd_moment = 0;
+
+    real_type pol_angle_1st_moment = 0;
+    real_type pol_angle_2nd_moment = 0;
+
+    for ([[maybe_unused]] int i : range(num_samples))
+    {
+        Interaction result = interact(rng_engine);
+        this->sanity_check(result);
+
+        real_type dir_angle = dot_product(result.direction, this->direction());
+        dir_angle_1st_moment += dir_angle;
+        dir_angle_2nd_moment += dir_angle * dir_angle;
+
+        real_type pol_angle = dot_product(result.polarization,
+                                          this->photon_track().polarization());
+        pol_angle_1st_moment += pol_angle;
+        pol_angle_2nd_moment += pol_angle * pol_angle;
+    }
+
+    dir_angle_1st_moment /= num_samples;
+    dir_angle_2nd_moment /= num_samples;
+    pol_angle_1st_moment /= num_samples;
+    pol_angle_2nd_moment /= num_samples;
```

Comment:
We could also check the average number of RNG samples.

Reply:
Is there a way to query how many times an engine has been sampled, or would I need to make a wrapper?

Reply:
The `DiagnosticRngEngine` used in these tests is a wrapper that does that, so you should be able to call `rng_engine.count()` at the end to get it.